js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -935,3 +936,42 @@ export const LAUNCH_ASSET_CHECK_UPSTREAM_QUERY = gql`
     }
   }
 `;
+
+async function checkIfResourcesRequireConfig({
+  client,
+  jobName,
+  repoAddress,
+  requiredResourceKeys,
+}: {
+  client: ApolloClient<any>;
+  jobName: string;
+  repoAddress: RepoAddress;
+  requiredResourceKeys: string[];
+}) {
+  const resourceResult = await client.query<
+    LaunchAssetLoaderResourceQuery,
+    LaunchAssetLoaderResourceQueryVariables
+  >({
+    query: LAUNCH_ASSET_LOADER_RESOURCE_QUERY,
+    variables: {
+      pipelineName: jobName,
+      repositoryName: repoAddress.name,
+      repositoryLocationName: repoAddress.location,
+    },
+  });
+  const pipeline = resourceResult.data.pipelineOrError;
+  if (pipeline.__typename !== 'Pipeline') {
+    return {
+      error: pipeline.message,
+      requiredConfigPresent: false,
+    };
+  }
+
+  const resources = pipeline.modes[0]!.resources.filter((r) =>
+    requiredResourceKeys.includes(r.name),
+  );
+  return {
```

Comment:
The logic in this function is unchanged, just moved it here to give it a function name